### DISCUSSION TITLE
consecutive 5xx rule matches without any other 2xx or 3xx responses in between particular time frame then should trigger alert

### DISCUSSION THREAD

**Discussion by vishalk663** [jertel/elastalert2#1264] [Q&A] [answered]
Hi All,

I am trying to setup alerting using elastalert and I am trying to achieve below scenarios.

scenario 1: Send alert if there are 5 consecutive 5xx errors. (Which is working fine)

scenario 2: Do not send alert if we get any 2xx or success between 5 5xx errors.

How to do this in elasalert ?

**Comment by jertel** [comment, marked as answer]
I'm unaware of a method to search Elasticsearch for this. I suggest asking the Elasticsearch community if this is possible first.

**Reply by vishalk663** [reply]
Thank you for the reply. 

we asked your question in elasticsearch community. 

https://discuss.elastic.co/t/feasibility-to-send-alerts-only-if-consecutive-errors-are-occurred-using-elastalert/341485

**Reply by jertel** [reply]
Your followup question you posted is different from the original question:

```
Thank you for the reply

Is there any method in elasticsearch to search 3 consecutive 400 errors in specific timeframe without any other 200 responses.
```

The original question asks if it's possible to exclude matches for 5 consecutive 500's **when it's interrupted by** a 200: 

```
No match: 5 5 5 2 5 5
   Match: 2 5 5 5 5 5
   Match: 5 5 5 5 5 5
```

Yet the second question asks if it's possible to exclude matches  for 3 consecutive 400 **within a timeframe when at least one** 200 **exists**:

```
No Match: 4 4 2 4
No Match: 4 4 4 2
No Match: 2 4 4 4
   Match: 4 4 4 4
```

These are conflicting questions you're asking.

**Comment by vishalk663** [comment]
Thank you for the correct question framing.

I asked the same in elasticsearch community
